src/lib.rs

```diff
@@ -258,11 +296,13 @@ impl ShaderBuilder {
 	///		All includes will be relative to the parent directory of the root WGSL module.
 	/// 	Code is generated recursively with attention to `include` and `define` statements.
 	/// 	See "Examples" for more details on include and macro functionality.
-	pub fn new(source_path: &str) -> Result<Self, ex::io::Error> {
+	/// - `defines` - list of strings which indicate the valid defines in wgsl
+	pub fn new(source_path: &str, defines: Option<&[String]>) -> Result<Self, ex::io::Error> {
```

Comment:
What I meant in [this thread in the previous PR](https://github.com/elyshaffir/wgsl_preprocessor/pull/1/files#r1038361719) is that for now we won't break the existing interface and only support conditional compilation using `//!define` statements.

Receiving a new parameter (even if it is optional) is breaking functionality, and this is the reason that [this](https://github.com/elyshaffir/wgsl_preprocessor/issues/4) issue was created and belongs to the milestone of version 2.0, which is the next major version in which there will probably be other breaking changes.

To conclude, for now please stick to just changing only `load_shader_module` in a way that it keeps track of `//!define` statements in a vector of definitions (even for included modules, so be sure to pass the vector of defines down recursively) and if it encounters an `//!ifdef` statement it checks that vector of definitions and ignores lines accordingly.

Reply:
Sorry but this is somewhat hard to do since defines can be parsed from an imported file but a file may not be imported depending on a condition so I would have to loop over all files twice or alternatively import on already parsed conditions which would be very annoying.

I would personally wait for v2 to add this functionality.